connect/runtime/src/test/java/org/apache/kafka/connect/storage/OffsetStorageWriterTest.java

```diff
@@ -192,6 +197,220 @@ public void testCancelAfterAwaitFlush() throws Exception {
         flushFuture.get(1000, TimeUnit.MILLISECONDS);
     }
 
+    @Test
```

Comment:
All these new tests seem to be specific to the `ConnectorOffsetBackingStore` class - should we move them to a new `ConnectorOffsetBackingStoreTest` class?

Reply:
I kept them here because eventually the error that should manifest in the callbacks should be via the `doFlush` method(which is what the actual `commitOffsets` method would be seeing. Let me know if that sounds ok.

Reply:
Alright. I created a new class for the same.

Reply:
Sorry, the point isn't just that we should have a new test suite with the same exact tests in it, it's that we should be testing the semantics of the `ConnectorOffsetBackingStore` class directly. We shouldn't be creating `OffsetStorageWriter` instances in these tests, we should be invoking, e.g., `ConnectorOffsetBackingStore::set` directly.

Reply:
Aah really sorry I missed that bit. I have updated the new test class to not have any traces of `OffsetStorageWriter` in it.